### DISCUSSION TITLE
Is it possible to clear data stored in the global state directly from the deactivate() method of an extension?

### DISCUSSION THREAD

**Discussion by JoshwinThomasIBM** [microsoft/vscode-discussions#2455] [Extension Development QnA]
I’m clearing global state data with context.globalState.update('libertyDataNew', '') in the deactivate() method, and it seems to work initially. But after reloading the extension, the data reappears. However, clearing it from activate() works as expected, even after reloading the extension . So, is it possible to reliably clear global state data in the deactivate() method?

**Comment by alefragnani** [comment]
Hi @JoshwinThomasIBM ,

The `globalState` memento stores data independent of the current opened VS Code window/instance, which means that all opened windows shares/access the same object, and one can interfere on another.

During your tests, are you sure you only had one VS Code window opened, prior to the reload command. If yes, could you try out closing and reopening VS Code, instead of reloading the window.

BTW, this is what the documentation says about the `deactivate` method (https://code.visualstudio.com/api/get-started/extension-anatomy#extension-entry-file).

> The extension entry file exports two functions, activate and deactivate (...) deactivate gives you a chance to clean up before your extension becomes deactivated (...) _if an extension needs to perform an operation when VS Code is **shutting down** or the extension is disabled or uninstalled, this is the method to do so._

I wonder if reloading the window or simply closing the folder (which maintains an empty VS Code window) would call `deactivate`. Based on this paragraph, I would say it won't be.

Hope this helps

**Reply by olguzzar** [reply]
Hi @JoshwinThomasIBM, did the above help you out?
